numba/parfors/parfor.py

```diff
@@ -128,13 +128,35 @@ def min_parallel_impl(return_type, arg):
         def min_1(in_arr):
             return in_arr[()]
     elif arg.ndim == 1:
-        def min_1(in_arr):
-            numba.parfors.parfor.init_prange()
-            min_checker(len(in_arr))
-            val = numba.cpython.builtins.get_type_max_value(in_arr.dtype)
-            for i in numba.parfors.parfor.internal_prange(len(in_arr)):
-                val = min(val, in_arr[i])
-            return val
+        if isinstance(arg.dtype, (types.NPDatetime, types.NPTimedelta)):
+            if numpy_version >= (1, 18):
+                def min_1(in_arr):
+                    numba.parfors.parfor.init_prange()
+                    min_checker(len(in_arr))
+                    val = numba.cpython.builtins.get_type_max_value(in_arr.dtype)
+                    for i in numba.parfors.parfor.internal_prange(len(in_arr)):
+                        if np.isnat(in_arr[i]):
+                            return in_arr[i]
+                        val = min(val, in_arr[i])
+                    return val
+            else:
+                def min_1(in_arr):
+                    numba.parfors.parfor.init_prange()
+                    min_checker(len(in_arr))
+                    val = numba.cpython.builtins.get_type_max_value(in_arr.dtype)
+                    for i in numba.parfors.parfor.internal_prange(len(in_arr)):
+                        if np.isnat(in_arr[i]):
+                            continue
```

Comment:
This doesn't seem like it will work correctly if all the values are NaT. Have you tested this with Numpy version 1.17?